numpy/core/_internal.py

```diff
@@ -287,23 +287,38 @@ def _newnames(datatype, order):
         return tuple(list(order) + nameslist)
     raise ValueError("unsupported order value: %s" % (order,))
 
+if sys.version_info >= (3, 0):
```

Comment:
Sorry, the real version I liked was to just use six.. Which is actualyl incorrect, it is to use:
```
from numpy.compat import basestring
```
and then just acting as if you are in python 2, so you can remove this completly.